tests/files/tests.py

```diff
@@ -340,6 +341,37 @@ def test_file_move_overwrite(self):
         os.close(handle_a)
         os.close(handle_b)
 
+    def test_file_move_copystat_cifs(self):
+        handle_a, self.file_a = tempfile.mkstemp()
+        handle_b, self.file_b = tempfile.mkstemp()
+
+        with mock.patch('django.core.files.move.copystat') as mock_copystat:
+            # simulate a CIFS filesystem by having copystat throw a permission error with EPERM
+            # which should be ignored.
+            # You must set the a message when passing in errno to exception side effect or
+            # errno will be ignored.
+            mock_copystat.side_effect = PermissionError(errno.EPERM, 'message')
+            self.assertIsNone(file_move_safe(self.file_a, self.file_b, allow_overwrite=True))
+
+        os.close(handle_a)
+        os.close(handle_b)
+
+    def test_file_move_copystat_not_cifs(self):
+        handle_a, self.file_a = tempfile.mkstemp()
+        handle_b, self.file_b = tempfile.mkstemp()
+
+        with mock.patch('django.core.files.move.copystat') as mock_copystat:
```

Comment:
I think it would be better to move `side_effect` to the `mock.patch`, i.e.:
```python
error = PermissionError(errno.EACCES, 'message')
with mock.patch('django.core.files.move.copystat', side_effect=error):
```